File: Leetcode/arrays/581_Shortest Unsorted Continuous Subarray/Solution.java
class Solution {
    public int findUnsortedSubarray(int[] nums) {
        int[] copy = nums.clone();
        Arrays.sort(nums);
        int start = 0, end = nums.length-1;
        while(start<=end && nums[start]==copy[start]) start+=1;
        while(end>=0 && nums[end]==copy[end]) end-=1;
        return (end-start>=0)? end-start+1: 0;
    }
}

File: Leetcode/arrays/581_Shortest Unsorted Continuous Subarray/Solution.py
class Solution(object):
    def findUnsortedSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        origin = list(nums)
        list.sort(nums)
        res = [origin[i]-nums[i] for i in range(len(nums))]
        start, end = 0, len(res)-1
        while start <= end and res[start]==0: start+=1
        while end>=0 and res[end]==0: end -= 1
        return end-start+1 if end-start>=0 else 0
